### DISCUSSION TITLE
GuestBook error in chapter Setting up an Admin Backend

### DISCUSSION THREAD

**Discussion by danyk4** [symfony/symfony#47082] [Symfony Book: The Fast Track] [answered]
I did everything but with empty "function configureFields()" in CommentCrudController.php I've got an error while trying to add a new comment:

An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'conference_id' cannot be null

I've noticed that there was no field "conference name" in a comment creation form and understood that it is about connection with Conference field. 
So I solved it with adding function configureFields with code from symfony doc:

public function configureFields(string $pageName): iterable
    {
        return [
            IdField::new('id')->onlyOnIndex(),
            Field::new('author'),
            TextareaField::new('text'),
            EmailField::new('email'),
            DateTimeField::new('createdAt'),
            AssociationField::new('conference'),
        ];
    }

and it works.

Please tell me, maybe it was another way because I'm new to symfony.

**Comment by danyk4** [comment, marked as answer]
oh, sorry, I saw this info was below

**Comment by KarlsonComplete** [comment]
Hello, I do not quite understand how you solved the problem? I have about the same error. " An exception occurred while executing a query: SQLSTATE[23502]: Not null violation: 7 ERROR: null value in column "conference_id" of relation "comment" violates not-null constraint"

**Reply by danyk4** [reply]
Hi
if you'll do all the same as in doc it is working. My problem was slightly another than yours.
check everything carefully

**Comment by williamsconcepts** [comment]
mine also didnt work, any help on this ?

**Comment by Romchik38** [comment]
looks like this is bad book chapter. I got 2 error. One of them the same. 
If you not be able to create a good job - do not do it. We are are not monkeys which beat a keyboard.

**Comment by olberger** [comment]
AFAIU, there's a missing step, which relates to configuring the fields of the comment, i.e. having the conference reference displayed in the form (and list). 

The diff shown later show that configureFields() isn't commented, but AFAIU, currently, the make:admin:crud will generate a commented-out configureFields() method.

So I guess there should be an additional paragraph, right before the addition of Conference::_toString() which uncomments and changes the method to :
``` php
public function configureFields(string $pageName): iterable
    {
        return [
            AssociationField::new('conference'),
            TextField::new('author'),
            EmailField::new('email'),
            TextareaField::new('text'),
            TextField::new('photoFilename'),
            DateTimeField::new('createdAt')
        ];
    }
```
Which exhibits the need to add __toString to Conference... and then hopefuly gets rid of the PSQL constraint violation error. And then only, one can proceed to fine tuning the rest of the Admin CRUD as in current docs.

Hope this helps,
